lib/galaxy/model/store/__init__.py

```diff
@@ -323,12 +342,14 @@ def remap_objects(p, k, obj):
                         if not in_directory(temp_dataset_file_name, self.archive_dir):
                             raise MalformedContents(f"Invalid dataset path: {temp_dataset_file_name}")
 
-                    if not file_name or not os.path.exists(temp_dataset_file_name):
+                    discarded_data = self.import_options.discarded_data
+                    if not file_name or not os.path.exists(temp_dataset_file_name) or discarded_data is ImportDiscardedDataType.FORCE:
                         dataset_instance.state = dataset_instance.states.DISCARDED
-                        dataset_instance.deleted = True
-                        dataset_instance.purged = True
-                        dataset_instance.dataset.deleted = True
-                        dataset_instance.dataset.purged = True
+                        deleted = discarded_data == ImportDiscardedDataType.FORBID
```

Comment:
Here we would set `deleted` to `False` if the model store doesn't contain the data but discarded_data is anything but `forbid`, which seems wrong ?

Isn't this the correct thing to do ?
```suggestion
                        deleted = True
```

I probably don't understand how this is going to work, but having a dataset whose deleted and purged status is False, but there's no actual data in the object store seems problematic ?